ls1intum/Artemis
# `Quiz Exercises`: Improve quiz save performance

### Checklist
#### General



### Motivation and Context


Saving/submitting a quiz did a few DB queries that were not strictly required. This PR reduces this number back down to 0 for synchronized mode and 1 count query for the others.

### Steps for Testing

Prerequisites:
- 1 Instructor
- Students
- Quiz Exercises in the various modes

1. Participate in quiz as the students, including multiple times as one students by using multiple tabs as that student.
2. Check that saving and submitting works correctly and that students cannot submit multiple times.
3. Repeat for the different modes.

### Review Progress





#### Code Review
#### Manual Tests
